test-network-function/lifecycle/suite.go

```diff
@@ -334,21 +334,16 @@ func testPodsRecreation(env *config.TestEnvironment) {
 			env.ResetOc()
 			// drain node
 			drainNode(n.Name) // should go in this
+			// Ensure the node is uncordoned even if we have a failure in waitForAllDeploymentsReady
+			defer uncordonNode(n.Name)
 			for _, ns := range env.NameSpacesUnderTest {
 				waitForAllDeploymentsReady(ns, scalingTimeout, scalingPollingPeriod)
-				// verify deployments are ready again
-				_, notReadyDeployments = getDeployments(ns)
-				if len(notReadyDeployments) > 0 {
-					uncordonNode(n.Name)
-					ginkgo.Fail(fmt.Sprintf("did not create replicas when node %s is drained", n.Name))
-				}
 			}
 			uncordonNode(n.Name)
-
-			for _, ns := range env.NameSpacesUnderTest {
-				// wait for all deployment to be ready otherwise, pods might be unreacheable during the next discovery
-				waitForAllDeploymentsReady(ns, scalingTimeout, scalingPollingPeriod)
-			}
+		}
+		for _, ns := range env.NameSpacesUnderTest {
+			// wait for all deployment to be ready otherwise, pods might be unreacheable during the next discovery
+			waitForAllDeploymentsReady(ns, scalingTimeout, scalingPollingPeriod)
```

Comment:
Shouldn't this be part of the defer block?

Reply:
Not really, based on the comment it's a post-recovery action to make sure the environment is clean before moving on.

I guess we could remove it, since we're doing it already after draining each node, but in some cases it might be good to keep it in case some operator tries to re-distribute the load after enabling a node.